src/store.js

```diff
@@ -158,10 +158,16 @@ export class Store {
       assert(path.length > 0, 'cannot register the root module by using registerModule.')
     }
 
-    this._modules.register(path, rawModule)
-    installModule(this, this.state, path, this._modules.get(path))
-    // reset store to update getters...
-    resetStoreVM(this, this.state)
+    const moduleResolved = module => {
+      this._modules.register(path, module.default || module)
+      installModule(this, this.state, path, this._modules.get(path))
+        // reset store to update getters...
+      resetStoreVM(this, this.state)
+      return module
+    }
+
+    if (isPromise(rawModule)) return rawModule.then(moduleResolved)
+    return moduleResolved(rawModule)
```

Comment:
If you return a Promise, it will mean you will need to ensure that Vuex is used with Promises which may require pollyfils. This will ensure users don't need it if they are following standard flow